### ISSUE TITLE
Why sentiment "Muppets songs were so cool" is negative

### ISSUE THREAD

**Issue by mohanupadhyay** [dotnet/machinelearning-samples#22]
I tried with sentiment "Muppets songs were so cool"
and output was negative 
training data is same given in dataset

**Comment by CESARDELATORRE** [comment]
Usually you'll need a very large dataset of cases for a sentiment model work properly.
Could you elaborate further on the data you had when training the model?
Did you add "Muppets songs were so cool" as a sample for the training?

Other than that, the only muppets example in https://github.com/dotnet/machinelearning-samples/blob/master/datasets/sentiment-imdb-train.txt is “_The Songs Were The Best And The Muppets Were So Hilarious_.” which has a positive sentiment. 

If your question is asking why the model predicts the sentiment as negative for "_Muppets songs were so cool_", it is possible that the predicted label is different from what is in the training data. The model does not memorize the exact string, but represents it in some other way (if it is informative). So the text model might remember that “so hilarious” and “the best” were positive but maybe “the muppets” was in another example as negative? 

Basically, if the samples in your dataset had many cases including "so cool" as positive in addition to other cases with "muppets songs" as positive, it should predict correctly.

The dataset we're providing in the example is small so it is easy to run the model. But for model's accuracy, in general, please test with a much larger size of training datasets.

I'm closing the issue, but feel free to open it if you do further testing with a much larger dataset and you still think that it is not predicting correctly, ok?

Thanks for your feedback.
